src/core_atmosphere/CMakeLists.txt

```diff
@@ -0,0 +1,153 @@
+
+## Unused
+# physics/physics_wrf/
+set(_atm_core_physics_wrf_srcs
+        libmassv.F
+        module_bl_gwdo.F
+        module_bl_mynn.F
+        module_bl_ysu.F
+        module_cam_error_function.F
+        module_cam_shr_kind_mod.F
+        module_cam_support.F
+        module_cu_gf.mpas.F
+        module_cu_kfeta.F
+        module_cu_ntiedtke.F
+        module_cu_tiedtke.F
+        module_mp_kessler.F
+        module_mp_radar.F
+        module_mp_thompson.F
+        module_mp_thompson_cldfra3.F
+        module_mp_wsm6.F
+        module_ra_cam.F
+        module_ra_cam_support.F
+        module_ra_rrtmg_lw.F
+        module_ra_rrtmg_sw.F
+        module_ra_rrtmg_vinterp.F
+        module_sf_bem.F
+        module_sf_bep.F
+        module_sf_bep_bem.F
+        module_sf_mynn.F
+        module_sf_noah_seaice.F
+        module_sf_noah_seaice_drv.F
+        module_sf_noahdrv.F
+        module_sf_noahlsm.F
+        module_sf_noahlsm_glacial_only.F
+        module_sf_oml.F
+        module_sf_sfcdiags.F
+        module_sf_sfclay.F
+        module_sf_urban.F)
+list(TRANSFORM _atm_core_physics_wrf_srcs PREPEND physics/physics_wrf/)
+
+# diagnostics/
+set(_atm_core_diagnostics_srcs
+        convective_diagnostics.F
+        isobaric_diagnostics.F
+        mpas_atm_diagnostic_template.F
+        mpas_atm_diagnostics_manager.F
+        mpas_atm_diagnostics_utils.F
+        pv_diagnostics.F
+        soundings.F)
+list(TRANSFORM _atm_core_diagnostics_srcs PREPEND diagnostics/)
+
+# dynamics/
+set(_atm_core_dynamics_srcs
+        mpas_atm_boundaries.F
+        mpas_atm_iau.F
+        mpas_atm_time_integration.F)
+list(TRANSFORM _atm_core_dynamics_srcs PREPEND dynamics/)
+
+# utils/
+set(_atm_core_utils_srcs
+        atmphys_build_tables_thompson.F
+        build_tables.F)
+list(TRANSFORM _atm_core_utils_srcs PREPEND utils/)
+
+# core_atosphere
+set(_atm_core_srcs
+        mpas_atm_core.F
+        mpas_atm_core_interface.F
+        mpas_atm_dimensions.F
+        mpas_atm_threading.F)
+
+## Generated includes
+set(atm_core_inc
+        block_dimension_routines.inc
+        core_variables.inc
+        define_packages.inc
+        domain_variables.inc
+        namelist_call.inc
+        namelist_defines.inc
+        setup_immutable_streams.inc
+        structs_and_variables.inc)
+
+
+add_library(core_atmosphere ${_atm_core_srcs}
+                            ${_atm_core_physics_srcs}
+                            ${_atm_core_physics_wrf_srcs}
+                            ${_atm_core_diagnostics_srcs}
+                            ${_atm_core_dynamics_srcs})
+
+mpas_core_target(CORE atmosphere TARGET core_atmosphere INCLUDES ${atm_core_inc})
+
+target_compile_definitions(core_atmosphere PRIVATE DO_PHYSICS=1)
+target_compile_definitions(core_atmosphere PRIVATE mpas=1)
+
+#Get physics_wrf tables from MPAS-Data
+include(FetchContent)
+FetchContent_Declare(mpas_data
+                     GIT_REPOSITORY https://github.com/MPAS-Dev/MPAS-Data.git
```

Comment:
I think this may be a source of differences between 16 and 18 columns when reading VEGPARM.TBL.  MPAS-Dev/MPAS-Model must retrieve the *.TBL files from somewhere other than https://github.com/MPAS-Dev/MPAS-Data.git, because the VEGPARM.TBL located in the build directory of MPAS-Model is not the same as the version in that repo.  We need to be using the 18 column version (which appears "magically" as far I am concerned) in the build directory, not the version in MPAS-Dev/MPAS-Data.git.

Reply:
So the tag just needs to be changed to `v7.0` instead of `master` below.  Then the changes in `src/core_atmosphere/physics/mpas_atmphys_lsm_noahinit.F` can be reverted.

Reply:
Note that MPAS-Model Makefile build systems populates the `src/atmosphere/physics/physics_wrf/files` directory with this script which downloads those table files from `MPAS-Dev/MPAS-Data/master`:  
[MPAS-Dev/src/atmosphere/physics/checkout_data_file.sh](https://github.com/MPAS-Dev/MPAS-Model/blob/51d5624709f75182130a2fd94715e1bb13310172/src/core_atmosphere/physics/checkout_data_files.sh#L75)

I am exactly emulating that process here.  I just need to find where to get the newer version of the TBL files.

Reply:
@jjguerrette Just saw your second post.  Yes that is the answer.  I didn't see the v7.0 table tag, because there was no branch.

Reply:
Yes, I was confused about the absence of other branches as well.  Per Michael's email, it might be nice to search for the appropriate version in the GIT_TAG, but I do not know how the MPAS team plans to manage those tags in the future.  It could be worth asking about.